### PR TITLE
ref(search) Remove unused code in saved searches

### PR DESCRIPTION
**Pull request by markstory**
Earlier this week I added a warning to see if any customers are using the project level saved searches. It hasn't captured any data which means this code is safe to remove.

I've slightly change the semantics on the project_group_index endpoint but I don't think that will matter much as that endpoint is also generally unused.

### CODE REVIEW

**Review by leedongwei — APPROVED** [review]
👏

File: src/sentry/models/savedsearch.py [hunk 1]
```diff
@@ -7,17 +7,6 @@
...
-    {"name": "Needs Triage", "query": "is:unresolved is:unassigned"},
-    {"name": "Assigned To Me", "query": "is:unresolved assigned:me"},
-    {"name": "My Bookmarks", "query": "is:unresolved bookmarks:me"},
-    {"name": "New Today", "query": "is:unresolved age:-24h"},
-    {"name": "Unhandled Errors", "query": "is:unresolved error.unhandled:true"},
-]
-DEFAULT_SAVED_SEARCH_QUERIES = set(search["query"] for search in DEFAULT_SAVED_SEARCHES)
-
-
 class SavedSearch(Model):
```
Comment (wedamija) [inline src/sentry/models/savedsearch.py:10]: I think we can also remove the `project` column and any related rows now that this is no longer used. Same for `is_default`, it looks like both those cols were only used for sentry 9

**Review by wedamija — APPROVED** [review]

**Comment by wedamija** [comment]
Looks like there are a couple uses of `DEFAULT_SAVED_SEARCHES` still around

File: src/sentry/models/savedsearch.py (hunk 1 above)
Comment (markstory) [reply src/sentry/models/savedsearch.py:10]: I was going to remove those columns once this had merged.

**Pull request merged**